internal/resolution/variablesources/bundle_test.go

```diff
@@ -4,19 +4,196 @@ import (
+			},
+
+			// Second level dependency that matches requirements of the first level dependency.
+			"second-level-dependency.v1.0.1": {
+				Bundle: declcfg.Bundle{
+					Name:    "second-level-dependency.v1.0.1",
+					Package: "second-level-dependency",
+					Properties: []property.Property{
+						{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "second-level-dependency", "version": "1.0.1"}`)},
+					},
+				},
+				CatalogName: fakeCatalogName,
+				InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+			},
+
+			// Second level dependency that does not match requirements of the first level dependency.
+			"second-level-dependency.v2.0.0": {
+				Bundle: declcfg.Bundle{
+					Name:    "second-level-dependency.v2.0.0",
+					Package: "second-level-dependency",
+					Properties: []property.Property{
+						{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "second-level-dependency", "version": "2.0.0"}`)},
+					},
+				},
+				CatalogName: fakeCatalogName,
+				InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+			},
+
+			// Package that is in a our fake catalog, but is not involved
+			// in this dependency chain. We need this to make sure that
+			// the testable function filters out unrelated bundles.
+			"uninvolved-package.v1.0.0": {
+				Bundle: declcfg.Bundle{
+					Name:    "uninvolved-package.v1.0.0",
+					Package: "uninvolved-package",
+					Properties: []property.Property{
+						{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "uninvolved-package", "version": "1.0.0"}`)},
+					},
+				},
+				CatalogName: fakeCatalogName,
+				InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+			},
+		}
+
+		allBundles := make([]*catalogmetadata.Bundle, 0, len(bundleSet))
+		for _, bundle := range bundleSet {
+			allBundles = append(allBundles, bundle)
+		}
+		requiredPackages := []*olmvariables.RequiredPackageVariable{
+			olmvariables.NewRequiredPackageVariable("test-package", []*catalogmetadata.Bundle{
+				bundleSet["first-level-dependency.v1.0.0"],
+			}),
+		}
+		installedPackages := []*olmvariables.InstalledPackageVariable{
+			olmvariables.NewInstalledPackageVariable("test-package", []*catalogmetadata.Bundle{
+				bundleSet["first-level-dependency.v1.0.0"],
+			}),
+		}
+
+		bundles, err := variablesources.MakeBundleVariables(allBundles, requiredPackages, installedPackages)
+		require.NoError(t, err)
+		require.Len(t, bundles, 3)
+
+		// Each dependency must have a variable.
+		// Dependencies from the same package must be sorted by version
+		// with higher versions first.
+		expectedIDs := []deppy.Identifier{
+			"fake-catalog-first-level-dependency-first-level-dependency.v1.0.0",
+			"fake-catalog-second-level-dependency-second-level-dependency.v1.0.1",
+			"fake-catalog-second-level-dependency-second-level-dependency.v1.0.0",
+		}
+		actualIDs := collectVariableIDs(bundles)
+		assert.Equal(t, expectedIDs, actualIDs)
```

Comment:
Need to check how tricky it is going to be to do something similar to #497 with `go-cmp` and `AllowUnexported`.